### PR TITLE
:art: Add PPM image output

### PR DESCRIPTION
**Pull request by micheltakken**
## This PR updates

The output of the cfd solvers. It adds a .ppm image format output option for result fields.

## PR Checklist
- [x] Did you include comments/documentation?
- [ ] Did you include and pass unit tests?

### CODE REVIEW

File: src/simulation/Simulation.h [hunk 1]
```diff
@@ -147,6 +147,7 @@ class Simulation {
     int continuousPhase = 0;                                                            ///< Fluid of the continuous phase.
     int iteration = 0;
     int maxIterations = 1e5;
+    int globalSteps = 0;
```
Comment (micheltakken) [inline src/simulation/Simulation.h:150]: Remove unused variable

File: src/simulation/Simulation.h [hunk 2]
```diff
@@ -599,6 +600,10 @@ class Simulation {
      */
     result::SimulationResult<T>* getSimulationResults();
 
+    std::tuple<T, T> getGlobalPressureBounds();
```
Comment (micheltakken) [inline src/simulation/Simulation.h:603]: add comments

File: src/simulation/Simulation.h [hunk 3]
```diff
@@ -629,6 +634,10 @@ class Simulation {
      * @brief Print the results as pressure at the nodes and flow rates at the channels
      */
     void printResults();
+
```
Comment (micheltakken) [inline src/simulation/Simulation.h:637]: Add comments

File: src/simulation/Simulation.hh [hunk 4]
```diff
@@ -593,6 +625,9 @@ namespace sim {
 
             }
 
+            writePressurePpm(getGlobalPressureBounds());
```
Comment (micheltakken) [inline src/simulation/Simulation.hh:628]: Add an if statement for this writing `if (writePpm)`

File: src/simulation/Simulation.hh [hunk 5]
```diff
@@ -1256,4 +1291,18 @@ namespace sim {
...
+    void Simulation<T>::writePressurePpm(std::tuple<T, T> bounds, int resolution) {
+        for (auto& [key, simulator] : cfdSimulators) {
+            simulator->writePressurePpm(0.98*std::get<0>(bounds), 1.02*std::get<1>(bounds), resolution);
+        }
+    }
+
+    template<typename T>
+    void Simulation<T>::writeVelocityPpm(std::tuple<T, T> bounds, int resolution) {
+        for (auto& [key, simulator] : cfdSimulators) {
+            simulator->writeVelocityPpm(0.98*std::get<0>(bounds), 1.02*std::get<1>(bounds), resolution);
```
Comment (micheltakken) [inline src/simulation/Simulation.hh:1304]: add comments explaining 0.98 and 1.02 values

File: src/simulation/simulators/cfdSimulator.h [hunk 6]
```diff
@@ -269,6 +269,34 @@ class CFDSimulator {
         throw std::runtime_error("The function prepareLattice is undefined for this CFD simulator.");
     }
     
+    /**
```
Comment (micheltakken) [inline src/simulation/simulators/cfdSimulator.h:272]: Adapt comments

File: src/simulation/simulators/cfdSimulator.h [hunk 7]
```diff
@@ -269,6 +269,34 @@ class CFDSimulator {
...
+
+    /**
+     * @brief Write the vtk file with results of the CFD simulation to file system.
+     * @param[in] iT Iteration step.
+    */
+    virtual void writeVelocityPpm (T min, T max, int imgResolution=600)
+    {
+        throw std::runtime_error("The function prepareLattice is undefined for this CFD simulator.");
+    }
+
```
Comment (micheltakken) [inline src/simulation/simulators/cfdSimulator.h:289]: Add comments

File: src/simulation/simulators/olbContinuous.h [hunk 8]
```diff
@@ -184,7 +184,11 @@ using BounceBack = olb::BounceBack<T,DESCRIPTOR>;
      * @brief Write the vtk file with results of the CFD simulation to file system.
      * @param[in] iT Iteration step.
     */
-    void writeVTK(int iT);
+    void writeVTK(int iT) override;
+
```
Comment (micheltakken) [inline src/simulation/simulators/olbContinuous.h:188]: Add comments

**Review by MariaEmmerich — APPROVED** [review]

**Pull request merged**